### PR TITLE
BUG: Fix PageRange.__add__ crashing with TypeError when start or stop is None

### PR DESCRIPTION
**Pull request by gaoflow**
## Problem

`PageRange.__add__` raises `TypeError: '>' not supported between instances of 'int' and 'NoneType'` (or `NoneType` and `int`, or `NoneType` and `NoneType`) whenever either operand has a `None` start or `None` stop, which is perfectly normal for ranges like `"5:"` or `":3"`.

```python
>>> from pypdf.pagerange import PageRange
>>> PageRange("0:5") + PageRange("3:")   # stop is None
TypeError: '>' not supported between instances of 'NoneType' and 'int'

>>> PageRange(":5") + PageRange(":3")    # both starts are None
TypeError: '>' not supported between instances of 'NoneType' and 'NoneType'
```

The problem is that `__add__` compares `slice.start` and `slice.stop` with `>` directly. `None` bounds are not comparable to integers, but they are semantically well-defined: `None` start = "beginning of sequence" and `None` stop = "end of sequence".

## Fix

Add two small helper functions inside `__add__` that map `None` to float sentinels before any comparison:

- `_start_key(None)` → `-inf`  (beginning sorts before any page index)
- `_stop_key(None)` → `+inf`   (end sorts after any page index)

Replace the direct comparisons and `max()` call with these key functions. No public API changes.

## Behaviour after fix

```python
>>> PageRange("0:5") + PageRange("3:")   # => PageRange("0:")
>>> PageRange(":5")  + PageRange(":3")   # => PageRange(":5")
>>> PageRange("5:")  + PageRange(":3")   # => ValueError: gap (correct)
>>> PageRange(":")   + PageRange("2:7")  # => PageRange(":")
```

All 23 existing tests continue to pass.

---

This pull request was prepared with the assistance of AI, under my direction and review.

### CODE REVIEW

File: pypdf/pagerange.py [hunk 1]
```diff
@@ -145,22 +145,28 @@ def __add__(self, other: "PageRange") -> "PageRange":
...
+        return PageRange(slice(a[0], stop))
 
 
 PAGE_RANGE_ALL = PageRange(":")  # The range of all pages.
 
 
-def parse_filename_page_ranges(
-    args: list[Union[str, PageRange, None]]
-) -> list[tuple[str, PageRange]]:
+def parse_filename_page_ranges(args: list[Union[str, PageRange, None]]) -> list[tuple[str, PageRange]]:
```
Comment (stefan6419846) [inline pypdf/pagerange.py]: Please avoid unnecessary changes.

File: pypdf/pagerange.py [hunk 2]
```diff
@@ -179,9 +185,7 @@ def parse_filename_page_ranges(
     for arg in [*args, None]:
         if PageRange.valid(arg):
             if not pdf_filename:
-                raise ValueError(
-                    "The first argument must be a filename, not a page range."
-                )
+                raise ValueError("The first argument must be a filename, not a page range.")
```
Comment (stefan6419846) [inline pypdf/pagerange.py]: Please avoid unnecessary changes.

**Review by stefan6419846 — CHANGES_REQUESTED** [review]

File: pypdf/pagerange.py (hunk 1 above)
Comment (binggao1230) [reply pypdf/pagerange.py]: Restored in `3a1e152a`. Verification passed:

```bash
python -m pytest tests/test_pagerange.py -q
python -m ruff check pypdf/pagerange.py tests/test_pagerange.py
```

File: pypdf/pagerange.py (hunk 2 above)
Comment (binggao1230) [reply pypdf/pagerange.py]: Restored in `3a1e152a`. Verification passed:

```bash
python -m pytest tests/test_pagerange.py -q
python -m ruff check pypdf/pagerange.py tests/test_pagerange.py
```

**Comment by binggao1230** [comment]
Reverted both unrelated reformatting changes (the `parse_filename_page_ranges` signature and the `raise ValueError` line) in 3a1e152a — the diff is now limited to the `__add__` None-bounds fix. Ready for another look.

File: pypdf/pagerange.py [hunk 3]
```diff
@@ -145,14 +145,22 @@ def __add__(self, other: "PageRange") -> "PageRange":
...
+
+        def _stop_key(v: Optional[int]) -> float:
+            return float("inf") if v is None else v
+
+        if _start_key(a[0]) > _start_key(b[0]):
             a, b = b, a
 
-        # Now a[0] is the smallest
-        if b[0] > a[1]:
+        # Now a has the smaller (or equal) start.
```
Comment (stefan6419846) [inline pypdf/pagerange.py]:
```suggestion
        # Now `a` has the smaller (or equal) start.
```

**Review by stefan6419846 — APPROVED** [review]

**Review by stefan6419846 — CHANGES_REQUESTED** [review]
Shouldn't we add some additional tests for this edge case?

**Comment by binggao1230** [comment]
Applied the suggestion (backticked `a`). Also added a dedicated `test_addition_none_bounds` parametrized case (6 combinations: None start absorbing, None stop absorbing, fully-open + bounded, fully-open + fully-open) plus `test_addition_gap_with_none_start` to confirm a None start doesn't mask a real gap. 30/30 green.

**Review by stefan6419846 — APPROVED** [review]
